### PR TITLE
WIP: Minio chart

### LINKED ISSUE #1814: minio: Helm chart support
Feature Request
=
Users should be able to install and configure Minio through a helm chart, similar to the existing support for Ceph: https://rook.io/docs/rook/master/helm-operator.html

### PR DESCRIPTION
**Pull request by sebastien-prudhomme**
**Description of your changes:**

An Helm chart for the Minio operator and another one for Minio object stores.

**Which issue is resolved by this Pull Request:**
Resolves #1814

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.

### CODE REVIEW

**Comment by sebastien-prudhomme** [comment]
Waiting for #1861 before finishing work of the objectstore chart.
The minio operator chart is working fine.

**Comment by galexrt** [comment]
@sebastien-prudhomme Sorry for no review on this. It seems that you merged branches into your PR branch, could you please rebase the changes on the latest `master` branch (there have been small changes in the values of the Minio CRD objects structure).
Then I'll gladly take a look so that we get this in fast.

**Comment by sebastien-prudhomme** [comment]
@galexrt There is still the problem with the fixed app label #1861 (the issue was closed by the bot)

**Comment by galexrt** [comment]
To note here: #2463 has been merged. Minio Object stores will now have a label added to them with the instance/object name (in `master` from now on).

File: cluster/charts/rook-minio-objectstore/templates/objectstore.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    nodeCount: 4
+  placement:
+    tolerations:
+    nodeAffinity:
+    podAffinity:
+    podAnyAffinity:
+  port: {{ .Values.service.port }}
+  credentials:
+    name: {{ template "rook-minio-objectstore.fullname" . }}
+  storageAmount: "10G"
```
Comment (galexrt) [inline cluster/charts/rook-minio-objectstore/templates/objectstore.yaml]: This is now a PVC list (https://rook.io/docs/rook/v0.9/minio-object-store-crd.html#sample) so the user can put one or more PVCs.
This should also be configurable by the user.

File: cluster/charts/rook-minio-objectstore/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app: {{ template "rook-minio-objectstore.name" . }}
+    release: {{ .Release.Name }}
```
Comment (galexrt) [inline cluster/charts/rook-minio-objectstore/templates/service.yaml]: Finally with https://github.com/rook/rook/pull/2463 merged, a selector to `objectstore: {{ template "rook-minio-objectstore.fullname" . }}` can be added here to be able to run multiple Minio instances in one namespace.

File: cluster/charts/rook-minio-objectstore/values.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Default values for rook-minio-objectstore.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: nginx
+  tag: stable
+  pullPolicy: IfNotPresent
```
Comment (galexrt) [inline cluster/charts/rook-minio-objectstore/values.yaml]: This section isn't needed, as there is no version, image option in the CRD (right now).

**Comment by travisn** [comment]
@sebastien-prudhomme are you still planning on working through this? Thanks!

**Comment by sebastien-prudhomme** [comment]
@travisn not have the time and the motivation to work on it. So i prefer to close it.

**Pull request closed without merging**